common/include/common/fundamentals/concepts.hpp

```diff
@@ -0,0 +1,11 @@
+#ifndef RDMN_CONCEPTS_HPP
+#define RDMN_CONCEPTS_HPP
+
+#include <concepts>
+
+namespace rdmn::core {
+template <typename T>
+concept NumberValue = std::integral<std::decay_t<T>> || std::floating_point<std::decay_t<T>>;
```

Comment:
I'd suggest replace `std::decay_t<T>` in both cases, with `std::remove_reference_t<std::remoce_cv_t<T>>`.